subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -135,6 +135,15 @@ void invalidate_public_key(uint32_t key_idx)
 	}
 }
 
+/*
+ * Don't include functions for reading and writing the counter when
+ * the counter has been disabled as this can only happen for corrupt
+ * configurations.
+ *
+ * We would rather have a link error in the build, than the API
+ * returning 0 at runtime for such builds.
+ */
+#ifdef CONFIG_SB_MONOTONIC_COUNTER
```

Comment:
Have you verified that the linker won't disregard these if you don't have the `ifdefs` ? I would expect the linker handle this.

Reply:
This ifdef is not to reduce code size, but to provoke a linker error when a user invokes the API without having enabled CONFIG_SB_MONOTONIC_COUNTER.

It doesn't make sense to call these functions while there are no counters and we want a linker error to prevent such a build from occuring.

Reply:
(if such a build did occur, the functions would definitely not be optimized away as they ultimately read volatile memory)

Reply:
I understand, the difference here compared to other use cases is that we have this API as part of a file which also exposes other APIs that can still be enabled.

Reply:
exactly

in hindsight these should have been two different libraries, and I have some patches that re-organizes them as such,
but due to them being provisioned together, and partitioned together, it is quite a big change, for (IMHO) not enough value, to separate them.